interface/src/app/gene/searchMany/searchMany.js

```diff
@@ -36,226 +36,151 @@ angular.module('adage.gene.searchMany', [
-            var query = data[i].search;
-            if (!searchResults[query]) {
-              // If search term didn't already exist
-              searchResults[query] = data[i]; // add it to the results
-              queries.push(query); // add to the list of queries
-            }
-          }
-          if (previousQueries !== queries.length) {
-            $rootScope.$broadcast('results.update');
-          }
-          searchSuccess = true;
-          $rootScope.$broadcast('results.searchResultsReturned');
-        },
-        function(responseObject, responseHeaders) {
-          searchSuccess = false;
-          $rootScope.$broadcast('results.searchResultsReturned');
-        }
-      );
-    }
+    templateUrl: 'gene/searchMany/gene-search-panel.tpl.html'
   };
 }])
 
 
 // Directive for whole gene search form
-.directive('geneSearchForm', ['SearchResults', function(SearchResults) {
+.directive('geneSearchForm', [function() {
   return {
-    controller: ['$scope', '$rootScope', 'SearchResults',
-      function($scope, $rootScope, SearchResults) {
-        $scope.loadingSearchResults = false;
+    controller: ['$scope', 'Gene', function($scope, Gene) {
+      $scope.errors = null;
+
+      $scope.searchGenes = function() {
+        // Gene.search will query for all of the search terms in qparams
+        // returning a list of objects containing each search term and the
+        // gene results for that term in each objects. If the search
+        // contained terms already found in our cache of searchResults,
+        // those results are ignored while terms not already present will
+        // be added to the cache.
+
+        if (!$scope.geneQueries) { // if the query is empty
+          return false;
+        }
 
-        // Clear any existing search results
-        SearchResults.clear();
-        $scope.errors = null;
+        var qparams = {'query': $scope.geneQueries};
 
-        $scope.searchGenes = function() {
-          if (!$scope.genesToAdd) { // if the query is empty
-            return false;
-          }
-          $scope.loadingSearchResults = true;
-
-          $rootScope.$broadcast('results.loadingSearchResults');
-
-          var qparams = {
-            'query': $scope.genesToAdd.query
-          };
-
-          if ($scope.organism) {
-            // TODO: adage doesn't currently support multiple organisms.
-            // Therefore, when retrieving gene objects from the API gene
-            // search endpoint, this will return *all* gene objects it finds,
-            // without filtering for organism. At the point when we add
-            // genes for more than one species to the database, we will
-            // need to set this $scope.organism to whatever organism is in
-            // the ML model (or genes for many organisms will be returned).
-            qparams['organism'] = $scope.organism;
-          }
-          SearchResults.search(qparams);
-        };
+        $scope.loadingSearchResults = true;
```

Comment:
Hah! What's wrong with `foobar`? :-)  I once had a classmate who liked to use the number `17` as a variable in his calculus problems.

On this particular variable name, I'd say that `loadingSearchResults` is a good description of state. A function that did that would be called something like `loadSearchResults`. A subtle distinction, perhaps, but the difference is very clear to me.

Something in between such as `resultsLoading` would be shorter but just as clear, in my opinion.